### PR TITLE
NIAD-2364: Necessary code changes and changes to expected test values.

### PR DESCRIPTION
**Pull request by dfk97**
Currently when given a descriptionDisplay value in code.coding.extensions that is not a valid text value against the given concept code, the PS adaptor is adding the default description registered to the snomed code. If the text does not match a SNOMED description then it should not add the descriptionDisplay.

### CODE REVIEW

File: gp2gp-translator/src/integrationTest/resources/json/LargeMessage/expectedBundleScenario6.json [hunk 1]
```diff
@@ -571,21 +546,14 @@
       "status": "final",
       "code": {
         "coding": [ {
-          "extension": [ {
-            "url": "https://fhir.nhs.uk/STU3/StructureDefinition/Extension-coding-sctdescid",
-            "extension": [ {
-              "url": "descriptionId",
-              "valueId": "254076013"
```
Comment (SRAlexander) [inline gp2gp-translator/src/integrationTest/resources/json/LargeMessage/expectedBundleScenario6.json:578]: This is another one that seems to be a valid code for the given text

**Review by SRAlexander — CHANGES_REQUESTED** [review]
Looks like there's an issue with some of the data, I'll re-review once corrected

File: gp2gp-translator/src/integrationTest/resources/json/expectedBundle.json [hunk 2]
```diff
@@ -23949,20 +23993,18 @@
           "extension": [ {
             "url": "https://fhir.nhs.uk/STU3/StructureDefinition/Extension-coding-sctdescid",
             "extension": [ {
-              "url": "descriptionId",
-              "valueId": "127639017"
```
Comment (SRAlexander) [inline gp2gp-translator/src/integrationTest/resources/json/expectedBundle.json:23953]: This is also a valid description id and should be present

File: gp2gp-translator/src/integrationTest/resources/e2e-mapping/output-json/PWTP10-output.json [hunk 3]
```diff
@@ -2061,27 +2061,27 @@
...
     "resource": {
       "resourceType": "Location",
-      "id": "{{LOCATION-2}}",
+      "id": "799FE127-B0C2-4D69-8781-CAC42310FD00",
       "meta": {
         "profile": [ "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-Location-1" ]
       },
       "identifier": [ {
         "system": "https://PSSAdaptor/B83002",
-        "value": "{{LOCATION-2}}"
```
Comment (SRAlexander) [inline gp2gp-translator/src/integrationTest/resources/e2e-mapping/output-json/PWTP10-output.json:2070]: We need these tags to remain the same

File: gp2gp-translator/src/integrationTest/resources/e2e-mapping/output-json/PWTP10-output.json [hunk 4]
```diff
@@ -2470,20 +2480,18 @@
           "extension": [ {
             "url": "https://fhir.nhs.uk/STU3/StructureDefinition/Extension-coding-sctdescid",
             "extension": [ {
-              "url": "descriptionId",
-              "valueId": "64172013"
```
Comment (SRAlexander) [inline gp2gp-translator/src/integrationTest/resources/e2e-mapping/output-json/PWTP10-output.json:2474]: ID matches the description text, this should be present

File: gp2gp-translator/src/integrationTest/resources/e2e-mapping/output-json/PWTP10-output.json [hunk 5]
```diff
@@ -2835,20 +2869,18 @@
           "extension": [ {
             "url": "https://fhir.nhs.uk/STU3/StructureDefinition/Extension-coding-sctdescid",
             "extension": [ {
-              "url": "descriptionId",
-              "valueId": "249995012"
```
Comment (SRAlexander) [inline gp2gp-translator/src/integrationTest/resources/e2e-mapping/output-json/PWTP10-output.json:2839]: Valid description id, it looks like we've got a pattern here in that we're losing descriptionId's against valid description text. Investigation required.

**Review by SRAlexander — CHANGES_REQUESTED** [review]

**Review by SRAlexander — APPROVED** [review]
Found cases all seem to be fixed, I've reviewed 6 of the test files with no issues found. That is a good indicator that we have covered all cases.

**Review by SRAlexander — DISMISSED** [review]

**Review by SRAlexander — APPROVED** [review]

File: gp2gp-translator/src/integrationTest/resources/e2e-mapping/output-json/PWTP10-output.json [hunk 6]
```diff
@@ -1275,19 +1275,6 @@
...
-              {
-                "url": "target",
-                "valueReference": {
-                  "reference": "Condition/03477CAD-F5F9-4058-8FB1-FB204DD589DA"
-                }
-              }
-            ]
-          }
-        ],
         "status": "current",
```
Comment (benhession) [inline gp2gp-translator/src/integrationTest/resources/e2e-mapping/output-json/PWTP10-output.json:1278]: Why is the related problem removed here?

File: gp2gp-translator/src/integrationTest/resources/e2e-mapping/output-json/PWTP10-output.json [hunk 7]
```diff
@@ -2635,6 +2626,8 @@
               "code": "78975002",
               "display": "Malignant essential hypertension"
             }
+          ],
+          "text": "Malignant essential hypertension"
           ]
```
Comment (benhession) [inline gp2gp-translator/src/integrationTest/resources/e2e-mapping/output-json/PWTP10-output.json:2631]: should there be an extra bracket here?

File: gp2gp-translator/src/integrationTest/resources/e2e-mapping/output-json/PWTP10-output.json [hunk 8]
```diff
@@ -2714,6 +2707,8 @@
               "code": "319775004",
               "display": "Aspirin 75mg tablets"
             }
+          ],
+          "text": "Aspirin 75mg tablets"
           ]
```
Comment (benhession) [inline gp2gp-translator/src/integrationTest/resources/e2e-mapping/output-json/PWTP10-output.json:2712]: Extra bracket again.

File: gp2gp-translator/src/integrationTest/resources/e2e-mapping/output-json/PWTP11-output.json [hunk 9]
```diff
@@ -2600,6 +2604,8 @@
               "code": "40055000",
               "display": "Chronic sinusitis"
             }
+          ],
+          "text": "Chronic sinusitis"
           ]
```
Comment (benhession) [inline gp2gp-translator/src/integrationTest/resources/e2e-mapping/output-json/PWTP11-output.json:2609]: bracket

**Review by SRAlexander — APPROVED** [review]

**Pull request merged**